internal/validate/kube/kube.go

```diff
@@ -1,389 +1,441 @@
 func Validate(ctx context.Context, clientSet *kubernetes.Clientset, restConfig *rest.Config, opts ...Option) error {
-	config := &Config{
-		namespace:  "default",
-		output:     os.Stdout,
-		exitStatus: false,
-		clientSet:  clientSet,
-		restConfig: restConfig,
-	}
-
-	for _, opt := range opts {
-		opt(config)
-	}
-
-	log.SetOutput(config.output)
-
-	var validations = []struct {
-		Validate   validation
-		WaitMsg    string
-		SuccessMsg string
-		ErrMsg     string
-	}{
-		{Pods, "validating pods", "pods validated", "validating pods failed"},
-		{Services, "validating services", "services validated", "validating services failed"},
-		{PVCs, "validating pvcs", "pvcs validated", "validating pvcs failed"},
-		{Connections, "validating connections", "connections validated", "validating connections failed"},
-	}
-
-	var totalFailCount int
-
-	for _, v := range validations {
-		log.Printf("%s %s...", validate.HourglassEmoji, v.WaitMsg)
-		results, err := v.Validate(ctx, config)
-		if err != nil {
-			return errors.Wrapf(err, v.ErrMsg)
-		}
-
-		var failCount int
-		var warnCount int
-		var succCount int
-
-		for _, r := range results {
-			switch r.Status {
-			case validate.Failure:
-				log.Printf("  %s failure: %s", validate.FailureEmoji, r.Message)
-				failCount++
-			case validate.Warning:
-				log.Printf("  %s warning: %s", validate.WarningSign, r.Message)
-				warnCount++
-			case validate.Success:
-				succCount++
-			}
-		}
-
-		if failCount > 0 || warnCount > 0 {
-			log.Printf("\n%s %s", validate.FlashingLightEmoji, v.ErrMsg)
-		}
-
-		if failCount > 0 {
-			log.Printf("  %s %d total failure(s)", validate.EmojiFingerPointRight, failCount)
-
-			totalFailCount = totalFailCount + failCount
-		}
-
-		if warnCount > 0 {
-			log.Printf("  %s %d total warning(s)", validate.EmojiFingerPointRight, warnCount)
-		}
-
-		if failCount == 0 && warnCount == 0 {
-			log.Printf("%s %s!", validate.SuccessEmoji, v.SuccessMsg)
-		}
-	}
-
-	if totalFailCount > 0 {
-		return errors.Newf("validation failed: %d failures", totalFailCount)
-	}
-
-	return nil
+    cfg := &Config{
+        namespace:  "default",
+        output:     os.Stdout,
+        exitStatus: false,
+        clientSet:  clientSet,
+        restConfig: restConfig,
+        eks:        false,
+    }
+
+    for _, opt := range opts {
+        opt(cfg)
+    }
+
+    log.SetOutput(cfg.output)
+
+    var validations []validation
+    validations = append(validations, validation{Pods, "validating pods", "pods validated", "validating pods failed"})
+    validations = append(validations, validation{Services, "validating services", "services validated", "validating services failed"})
+    validations = append(validations, validation{PVCs, "validating pvcs", "pvcs validated", "validating pvcs failed"})
+    validations = append(validations, validation{Connections, "validating connections", "connections validated", "validating connections failed"})
```

Comment:
nit: I think this would be more idiomatic Go
```suggestion
 validations := []validation{
    {Pods, "validating pods", "pods validated", "validating pods failed"},
    {Services, "validating services", "services validated", "validating services failed"},
    {PVCs, "validating pvcs", "pvcs validated", "validating pvcs failed"},
    {Connections, "validating connections", "connections validated", "validating connections failed"},
}
```